wgpu-hal/src/vulkan/adapter.rs

```diff
@@ -1834,7 +1835,40 @@ impl super::Adapter {
 
         let mem_allocator = {
             let limits = self.phd_capabilities.properties.limits;
-            let config = gpu_alloc::Config::i_am_prototyping(); //TODO
+            // TODO: These configuration options should take hardware capabilities
+            // into consideration.
+            let mb = 1024 * 1024;
+            let perf_cfg = gpu_alloc::Config {
+                starting_free_list_chunk: 128 * mb,
+                final_free_list_chunk: 512 * mb,
```

Comment:
`perf_cfg` seems to be equivalent to `Config::i_am_prototyping` except that `final_free_list_chunk` is 512 instead of 128. Any specific motivation around these numbers, how did they come to be?

Reply:
Some guesswork and a bit of looking at the vulkan memory allocator from AMD's SDK. But mostly guesswork. It's difficult to find good references about how to do this right.

`final_free_list_chunk` should be bigger than `starting_free_list_chunk` if we want the behavior of starting with smaller block sizes and using larger ones only after we observe that the small ones aren't enough, which I think is a good "I don't know what the workload is going to be like" approach.

For reference, `VMA`, and `gpu_allocator` both start with 256 MB blocks (then VMA doubles the block size each time it needs a new block). I kept the initial block size at 128 MB to avoid changing the current behavior too much without having a good sense of what the right parameters are and to limit risk. At some point it would be good to experiment with real workloads, perhaps with the help of folks in the bevy community, and get a better feel for what would make good defaults.

Reply:
👍 makes sense!
Leaving a comment of almost verbatim what you just wrote would be a good even if it boils down to "not quite sure what numbers to put here"